src/transport_performance/utils/defence.py

```diff
@@ -30,12 +30,16 @@ def _is_path_like(pth, param_nm):
 
 def _check_parent_dir_exists(pth, param_nm, create=False):
     _is_path_like(pth, param_nm)
+    # replace back slashes to ensure consistency
+    pth = pth.replace("\\", "/").replace(repr("\\"), "").replace("'", "")
```

Comment:
This isn't ideal but it's the only way to ensure paths are consistent and also that paths including singular \ (escape char) are correctly processed. It also provides the leniency for the users path to include _n_ number of '\\' in a path.  
An example of this includes `out_pth="D:\\\dir1\dir2\\dir3\dir4\file.html"`